### PR TITLE
Connecting applications documentation take 2

### PR DESCRIPTION
**Pull request by bprashanth**
Adds an https example and more cleanup.  Depends on some form of https://github.com/GoogleCloudPlatform/kubernetes/pull/11246  getting in.
@bgrant0607 @thockin

### CODE REVIEW

**Comment by bprashanth** [comment]
Shippable will fail till i fix the links that point to the nginx-https example

**Comment by thockin** [comment]
LGTM

File: docs/user-guide/connecting-applications.md [hunk 1]
```diff
@@ -145,89 +153,218 @@ kube-dns   <none>       k8s-app=kube-dns     10.0.0.10   53/UDP
...
 ```
 
-## Exposing the Service to the internet
+## Securing the Service
 
-For some parts of your applications you may want to expose a Service onto an external IP address. Kubernetes supports two ways of doing this: NodePorts and LoadBalancers.
-Lets recreate the nginx Service with a node port:
+Till now we have only accessed the nginx server from within the cluster. Before exposing the Service to the internet, you want to make sure the communication channel is secure. For this, you will need:
+* Self signed certificates for https
+* An nginx server configured to use the cretificates
```
Comment (nikhiljindal) [inline docs/user-guide/connecting-applications.md]: cretificates -> certificates

**Comment by erictune** [comment]
Can you rebase.

**Comment by bprashanth** [comment]
This can't go in till https://github.com/GoogleCloudPlatform/kubernetes/pull/11246 because the links are broken. Removing ok-to-merge till that gets it, to avoid confusion.

File: docs/user-guide/connecting-applications.md [hunk 2]
```diff
@@ -102,24 +110,36 @@ spec:
...
 nginxsvc     10.245.0.14:80,10.245.0.15:80
 ```
-You should now be able to curl the nginx Service on `10.0.208.159:80` from any node in your cluster. Note that the Service ip is completely virtual, it never hits the wire, if you’re curious about how this works you can read more about the [service proxy](services.md#virtual-ips-and-service-proxies).
+You should now be able to curl the nginx Service on `10.0.116.146:80` from any node in your cluster. Note that the Service ip is completely virtual, it never hits the wire, if you’re curious about how this works you can read more about the [service proxy](services.md#virtual-ips-and-service-proxies).
 
-## Accessing the Service from other pods in the cluster
+## Accessing the Service
 
-Kubernetes supports 2 primary modes of finding a Service - environment variables and DNS. The former works out of the box while the latter requires the [kube-dns cluster addon](../../cluster/addons/dns/README.md).
+Kubernetes supports 2 primary modes of service discovery - environment variables and DNS. The former works out of the box while the latter requires the [kube-dns cluster addon](../../cluster/addons/dns/README.md).
```
Comment (bgrant0607) [inline docs/user-guide/connecting-applications.md]: Technically, the previous text was more correct. Discovery is about detection of which services exist, such as pulling up a list of available printers on your network. We just do name resolution.

File: docs/user-guide/connecting-applications.md [hunk 3]
```diff
@@ -154,88 +174,218 @@ kube-dns   <none>       k8s-app=kube-dns     10.0.0.10   53/UDP
...
+$ curl https://104.197.63.17:30645 -k
+...
+<h1>Welcome to nginx!</h1>
+```
 
-$ kubectl get service nginxsvc -o json | grep \"ip\"
-"ip": "104.197.37.222"
+Lets now recreate the Service to use a cloud load balancer, just change the `Type` of Service in the nginx-app.yaml from `NodePort` to `LoadBalancer`:
+```shell
+$ kubectl delete rc, svc -l app=nginx; kubectl create -f nginx-app.yaml
```
Comment (bgrant0607) [inline docs/user-guide/connecting-applications.md]: Any particular reason why you put these on the same line?

**Comment by bgrant0607** [comment]
LGTM

**Comment by erictune** [comment]
Should be good to rebase now.

**Comment by bprashanth** [comment]
This is all green, @thockin can I reapply the ok-to-merge?

**Comment by bgrant0607** [comment]
ok to merge

**Comment by thockin** [comment]
LGTM

**Pull request merged**